### DISCUSSION TITLE
Open record in modal without viewaction

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#6830] [Help]
When I add a ViewAction to my table a new column is created that gives me the opportunity to open the record in a modal.
But I would rather like the row's click to trigger the modal popup and therefore omit the extra space consuming column.
Is this possible?

Thank you

**Comment by danharrin** [comment]
You cannot open a modal when the row is clicked without the corresponding Action object being in either `actions()` or `Column::make()->action()`

**Reply by OvalMedia** [reply]
meaning: I have to live with the view button being present in the table at all times?

**Reply by danharrin** [reply]
Yes, otherwise there wouldnt be a place to register it atm

**Reply by lucacastelnuovo** [reply]
@OvalMedia I wanted the same thing (https://github.com/filamentphp/filament/discussions/6552#discussioncomment-6228660). Dan's solution of calling `iconButton()` helped a bit with the button taking up too much space.

**Reply by OvalMedia** [reply]
Thx, that definitely saved up some space!

**Reply by ryanmortier** [reply]
I hit this today as well when using a ManageRelatedRecords page. Can we put in a feature request for this?
